sky/resources.py

```diff
@@ -51,6 +51,7 @@ def __init__(
         disk_size: Optional[int] = None,
         disk_tier: Optional[Literal['high', 'medium', 'low']] = None,
         ports: Optional[List[Union[int, str]]] = None,
+        specific_reservations: Optional[List[str]] = None,
```

Comment:
After discussing them internally, we are thinking that it might be better to use the global configuration file `~/.sky/config.yaml` to specify the `specific_reservations` as it is cloud-specific arguments.

We are thinking to have a field in the config file:
```
gcp:
    specific_reservations:
        - xxx/xxx
```

In the code, we can access that global configs with `skypilot_config.get_nested(('gcp', 'specific_reservations'), None)`.